docs/spec_draft.md

```diff
@@ -2126,6 +2128,87 @@ More formally, for each element `x`: `real(x) = is_complex(x) ? x.real : x`.
 
 [Back to Ops](#index-of-ops)
 
+## stablehlo.recv
+
+### Semantics
+
+Receives data to a partition from a [stablehlo.send](#stablehlosend)
+instruction, in another partition, that shares the same channel handle
+`channel_handle`.
+
+`stablehlo.recv`, paired with the corresponding [stablehlo.send](#stablehlosend)
+instruction, represents synchronous communication. However, in order to enable
+asynchronous data transfers, it is decomposed into two HLO instructions
+(`Recv` and `RecvDone`). Similarly, the paired
+[stablehlo.send](#stablehlosend) is internally decomposed into two HLO
+instructions (`Send` and `SendDone`).
```

Comment:
Such pairs of send/recv and *dones are generally useful. It would be good to expose such a capability on this level as well. I am not sure why this is mentioned here without being able to express it explicitly. As an implicit feature it is just an implementation/optimization detail.

Is there any particular reason for not providing SendDone and RecvDone on this level?

Reply:
The current version mirrors what we have in MHLO.  
We are planning to explore the option via https://github.com/openxla/stablehlo/issues/642.